src/store/slices/formSlice.ts

```diff
@@ -0,0 +1,40 @@
+import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
+import { type FormData } from '../../types';
+
+interface FormState {
+  submissions: FormData[];
+  newSubmissionId: string | null;
+}
+
+const initialState: FormState = {
+  submissions: [],
+  newSubmissionId: null,
+};
+
+export const formSlice = createSlice({
+  name: 'form',
+  initialState,
+  reducers: {
+    addSubmission: (state, action: PayloadAction<Omit<FormData, 'id'>>) => {
+      const newSubmission: FormData = {
+        ...action.payload,
+        id: Date.now().toString(),
```

Comment:
It's better to use `nanoid()` from `@reduxjs/toolkit` for generating unique IDs because it provides collision-resistant, cryptographically secure, and compact identifiers, which are more reliable than using timestamps like Date.now().